Framework/Algorithms/src/PolarizationCorrections/PolarizerEfficiency.cpp

```diff
@@ -0,0 +1,204 @@
+                  "Polarizer efficiency as a function of wavelength");
+
+  const auto &spinValidator = std::make_shared<SpinStateValidator>(std::unordered_set<int>{2, 3, 4});
+  declareProperty(PropertyNames::SPIN_STATES, "11,10,01,00", spinValidator,
+                  "Order of individual spin states in the input group workspace, e.g. \"01,11,00,10\"");
+
+  declareProperty(std::make_unique<FileProperty>(PropertyNames::OUTPUT_FILE_PATH, "", FileProperty::OptionalSave),
+                  "File name or path for the output to be saved to.");
+}
+
+/**
+ * Tests that the inputs are all valid
+ * @return A map containing the incorrect workspace
+ * properties and an error message
+ */
+std::map<std::string, std::string> PolarizerEfficiency::validateInputs() {
+  std::map<std::string, std::string> errorList;
+  // Check input workspaces.
+  const WorkspaceGroup_sptr inputWorkspace = getProperty(PropertyNames::INPUT_WORKSPACE);
+  if (inputWorkspace == nullptr) {
+    errorList[PropertyNames::INPUT_WORKSPACE] = "The input workspace is not a workspace group.";
+    return errorList;
+  }
+
+  auto const &inputWsCount = inputWorkspace->size();
+  if (inputWsCount < 2 || inputWsCount > 4) {
+    errorList[PropertyNames::INPUT_WORKSPACE] =
+        "The input group workspace must have at least two periods corresponding to the spin configurations.";
+  } else {
+    for (size_t i = 0; i < inputWsCount; ++i) {
+      const MatrixWorkspace_sptr stateWs = std::dynamic_pointer_cast<MatrixWorkspace>(inputWorkspace->getItem(i));
+      validateInputWorkspace(stateWs, PropertyNames::INPUT_WORKSPACE, errorList);
+    }
+  }
+  const MatrixWorkspace_sptr analyserWs = getProperty(PropertyNames::ANALYSER_EFFICIENCY);
+  validateInputWorkspace(analyserWs, PropertyNames::ANALYSER_EFFICIENCY, errorList);
+
+  const auto &spinConfigurationInput = getPropertyValue(PropertyNames::SPIN_STATES);
+  const auto &spinStateCount = PolarizationCorrectionsHelpers::splitSpinStateString(spinConfigurationInput).size();
+  if (spinStateCount != inputWsCount) {
+    errorList[PropertyNames::SPIN_STATES] =
+        "The number of workspaces in the input WorkspaceGroup ( " + std::to_string(inputWsCount) +
+        ") does not match the number of spin states provided (" + std::to_string(spinStateCount) + ").";
+  }
+  const auto &t01Ws = PolarizationCorrectionsHelpers::workspaceForSpinState(inputWorkspace, spinConfigurationInput,
+                                                                            SpinStateValidator::ZERO_ONE);
+  const auto &t00Ws = PolarizationCorrectionsHelpers::workspaceForSpinState(inputWorkspace, spinConfigurationInput,
+                                                                            SpinStateValidator::ZERO_ZERO);
+  if (t01Ws == nullptr || t00Ws == nullptr) {
+    errorList[PropertyNames::SPIN_STATES] = "The required spin configurations (00, 01) could not be found in the given"
```

Comment:
I unexpectedly get this as a second validation error message if I pass in spin configuration `00,01,10,11` with a group of table workspaces (the crash we were getting with this type of input workspace group is now resolved, though, so that's great). I think this might be resolved if we were able to use `PolarizationCorrectionsHelpers::indexOfWorkspaceForSpinState` directly in the validation here rather than `PolarizationCorrectionsHelpers::workspaceForSpinState`?